src/npctalk.cpp

```diff
@@ -5573,6 +5573,36 @@ talk_effect_fun_t::func f_die( bool is_npc )
     };
 }
 
+talk_effect_fun_t::func f_die_advanced( const JsonObject &jo, std::string_view member,
+                                        const std::string_view, bool is_npc )
+{
+    JsonObject job = jo.get_object( member );
+    bool remove_corpse = job.get_bool( "remove_corpse", false );
+    bool supress_message = job.get_bool( "supress_message", false );
+    bool remove_from_creature_tracker = job.get_bool( "remove_from_creature_tracker", false );
+
+    return [remove_corpse, supress_message, remove_from_creature_tracker,
+                   is_npc]( dialogue const & d ) {
+        map &here = get_map();
+
+        if( d.actor( is_npc )->get_monster() ) {
+            monster &mon = *d.actor( is_npc )->get_monster();
+            if( remove_from_creature_tracker ) {
+                get_creature_tracker().remove( mon );
+                return;
+            }
+            mon.death_drops = !remove_corpse;
```

Comment:
This could set death_drops to true even if the monster is not normally supposed to have them. For example see the `temporary_drop_items` field of `u_spawn_monster`.

One could make an eoc that spawns a monster and explicitly make it a temporarily-summoned monster that doesn't drop items, but then your function overrides it. This is very counter-intuitive and should probably not be the case.